### PR TITLE
add min order quantity error to product cart processor

### PR DESCRIPTION
**Pull request by NilsEvers**
### 1. Why is this change necessary?
When adding a product to the cart, with a lower quantity than the product _min order quantity_, the line items quantity will be silently changed to the _min order quantity_, without any errors. This is especially unfavourable when working with the API, as there is no indication that the quantity was updated. 
With this change, the behaviour is the same as when the quantity is updated to match the _purchase steps_.

### 2. What does this change do, exactly?
- Adds a new Cart Error `\Shopware\Core\Content\Product\Cart\MinOrderQuantityError`
- Refactors `\Shopware\Core\Content\Product\Cart\ProductCartProcessor::validateStock` to add an error to the cart when the quantity is modified to match the products min order quantity.
- And some minor refactoring of the validateStock method

### 3. Describe each step to reproduce the issue or behaviour.
via API
- add a product line item to the cart with a quantity of 1, where the min order quantity of the product is higher
- the quantity will be updated to match the min order quantity without any errors

### 4. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Content/Product/Cart/ProductCartProcessor.php [hunk 1]
```diff
@@ -234,11 +234,7 @@ private function validateStock(LineItem $item, Cart $cart, LineItemCollection $s
             $steps = $item->getQuantityInformation()->getPurchaseSteps() ?? 1;
         }
 
-        if ($item->getQuantity() < $minPurchase) {
-            $item->setQuantity($minPurchase);
-        }
-
-        if ($available <= 0 || $available < $minPurchase) {
```
Comment (NilsEvers) [inline src/Core/Content/Product/Cart/ProductCartProcessor.php:241]: $minPurchase is at least 1, so the left side of the condition is always true and can be removed

File: src/Core/Content/Product/Cart/MinOrderQuantityError.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public function getLevel(): int
+    {
+        return self::LEVEL_WARNING;
+    }
+
+    public function blockOrder(): bool
+    {
+        return true;
```
Comment (keulinho) [inline src/Core/Content/Product/Cart/MinOrderQuantityError.php:62]: Maybe change to `false` so the order process is not blocked

Comment (NilsEvers) [reply src/Core/Content/Product/Cart/MinOrderQuantityError.php:62]: I"ve based this on the the `PurchaseStepsError`, which is also blocking. In my opinion the two cases are quite similar and should be handled in the same way.

**Comment by keulinho** [comment]
Hey @NilsEvers thanks for your contribution, we've merged your changes! :tada: :blue_heart:

**Pull request closed without merging**